pkg/registry/core/pod/storage/eviction.go

```diff
@@ -226,6 +223,10 @@ func (r *EvictionREST) checkAndDecrement(namespace string, podName string, pdb p
 		return err
 	}
 
+	if _, ok := pdb.Status.DisruptedPods[podName]; ok {
```

Comment:
I think we want to do this check only for deciding whether PodDisruptionsAllowed should be decremented. The update of the timestamp for the pod that happens on line 244 should probably happen anyway so pods are no prematurely removed from the DisruptedPods map.